include/swift/Basic/Casting.h

```diff
@@ -41,9 +41,17 @@ Destination function_cast(Source source) {
   static_assert(std::is_trivially_copyable_v<Destination>,
                 "The destination type must be trivially constructible");
 
+#if __has_feature(ptrauth_calls)
+  // Use reinterpret_cast here so we perform any necessary auth-and-sign.
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wcast-function-type-mismatch"
```

Comment:
I don't think that we should be silencing away this warning. This is technically UB, and the compiler is pointing that out.

Reply:
I don’t see any use in leaving it. The cast is necessary for correct behavior when `ptrauth_calls` is on, so we’d just have a warning we can’t address.